### PR TITLE
[2.19.x] DDF-5897: Fixed drawing polygons without buffers

### PR DESCRIPTION
**Pull request by leo-sakh**
#### What does this PR do?
Fixes a bug introduced by #5851 

#### Who is reviewing it? 
@bdeining 
@millerw8 
@bennuttle 
@frnkshin 
@nsuvarna 
@lavoywj 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@millerw8

#### How should this be tested?
Same as #5801 

#### Any background context you want to provide?
See #5851 

#### What are the relevant tickets?
Fixes: #5897 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js [hunk 1]
```diff
@@ -50,19 +50,49 @@ const translateFromOpenlayersCoordinates = coords => {
...
 /**
- * Returns buffer width in meters with the lowest value of 1 meter.
+ * Returns buffer width in meters, if applicable, with the lowest value of 1 meter.
  */
 const getConstrainedBufferWidth = model => {
+  const polygonWidthValue = model.get('polygonBufferWidth')
+
+  if (!isValidBufferWidth(polygonWidthValue)) return undefined
+
+  if (0 == polygonWidthValue) return undefined
```
Comment (bennuttle) [inline ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js]: ✏️ This seems readable to me, but I know some folks don't like single-line `if`s without brackets.

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js (hunk 1 above)
Comment (Lambeaux) [reply ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js:64]: If this doesn't bother you guys then I won't make a fuss, but I personally don't like debugging through code and losing my current context without clearly seeing what the result of the `if` statement was. Sure you could infer it from your loss of context but that strikes me as error prone.

File: ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js [hunk 2]
```diff
@@ -50,19 +50,49 @@ const translateFromOpenlayersCoordinates = coords => {
...
+      'meters'
+    ).geometry.coordinates.map(set => {
+      return Turf.polygon([set])
+    })
+
+  return polySegment.geometry.coordinates.map(set => {
+    /**
+     * There appears to be a bug in the current version of Turf that produces 2-point arrays instead of 4
+     * which results in an exception
+     */
```
[the remark this answers is not in the document]
Comment (Lambeaux) [reply ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js:89]: +1

File: ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js (hunk 1 above)
Comment (leo-sakh) [reply ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js:64]: Sorry, @Lambeaux are you talking about the lack of `{}` here? I'm not quite following.

Comment (Lambeaux) [reply ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js:64]: Yeah I'm talking directly to the behavior of (at least the Java) debugger when returns are on the same line as control flow decisions.

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

**Comment by bennuttle** [comment]
LGTM, able to draw shapes without buffers and verified the intended limits are working correctly

**Pull request merged**